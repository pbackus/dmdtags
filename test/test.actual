!_TAG_FILE_SORTED	1	/0=unsorted, 1=sorted, 2=foldcase/
C	test/test.d	16;"	c	struct:S
E	test/test.d	23;"	g
I	test/test.d	12;"	i	struct:S
PrivateStruct	test/test.d	56;"	s	file:
S	test/test.d	10;"	s
ThreeInts	test/test.d	35;"	a
bar	test/test.d	41;"	v
c	test/test.d	17;"	v	struct:S
enumMember	test/test.d	25;"	e	enum:E
eponymous	test/test.d	44;"	T
eponymous	test/test.d	44;"	f
eponymousWithExtra	test/test.d	61;"	T
eponymousWithExtra	test/test.d	63;"	v
foo	test/test.d	40;"	v
fun	test/test.d	28;"	f
hidden	test/test.d	64;"	v
method	test/test.d	20;"	f	struct:S
overloaded	test/test.d	46;"	T
overloaded	test/test.d	48;"	f
overloaded	test/test.d	49;"	f
privateMember	test/test.d	58;"	v	struct:PrivateStruct	file:
privateVar	test/test.d	54;"	v	file:
stuff	test/test.d	36;"	v
temp	test/test.d	38;"	T
test	test/test.d	6;"	M
versionedSymbol	test/test.d	52;"	v
w	test/test.d	15;"	v	struct:S
x	test/test.d	8;"	v
y	test/test.d	19;"	v	struct:S
z	test/test.d	14;"	v	struct:S
