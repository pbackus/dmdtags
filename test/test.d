/++
Test module for dmdtags.

Author: Paul Backus
+/
module test;

int x;

struct S
{
	interface I {}
	union {
		int z;
		double w;
		class C : I {}
		C c;
	}
	int y;
	void method() {}
}

enum E
{
	enumMember
}

void fun()
{
	struct Local {}
}

import std.meta: AliasSeq;

alias ThreeInts = AliasSeq!(int, int, int);
ThreeInts stuff;

template temp(T)
{
	enum foo = T.sizeof;
	enum bar = T.stringof;
}

void eponymous(T)(T t) {}

template overloaded(T)
{
	T overloaded(int) {}
	T overloaded(double) {}
}

version (Foo) int versionedSymbol;

private int privateVar;

private struct PrivateStruct
{
	int privateMember;
}

template eponymousWithExtra()
{
	int eponymousWithExtra;
	int hidden;
}
