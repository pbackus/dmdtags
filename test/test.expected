!_TAG_FILE_SORTED	1	/0=unsorted, 1=sorted, 2=foldcase/
C	test/test.d	16;"	c	struct:S
E	test/test.d	23;"	g
I	test/test.d	12;"	i	struct:S
PrivateStruct	test/test.d	56;"	s	file:
S	test/test.d	10;"	s
ThreeInts	test/test.d	35;"	a
bar	test/test.d	41;"	v	template:temp
c	test/test.d	17;"	v	struct:S
enumMember	test/test.d	25;"	e	enum:E
eponymous	test/test.d	44;"	T	signature:(T)
eponymous	test/test.d	44;"	f	template:eponymous	signature:(T t)
eponymousWithExtra	test/test.d	61;"	T	signature:()
eponymousWithExtra	test/test.d	63;"	v	template:eponymousWithExtra
foo	test/test.d	40;"	v	template:temp
fun	test/test.d	28;"	f	signature:()
hidden	test/test.d	64;"	v	template:eponymousWithExtra
method	test/test.d	20;"	f	struct:S	signature:()
overloaded	test/test.d	46;"	T	signature:(T)
overloaded	test/test.d	48;"	f	template:overloaded	signature:(int)
overloaded	test/test.d	49;"	f	template:overloaded	signature:(double)
privateMember	test/test.d	58;"	v	struct:PrivateStruct	file:
privateVar	test/test.d	54;"	v	file:
stuff	test/test.d	36;"	v
temp	test/test.d	38;"	T	signature:(T)
test	test/test.d	6;"	M
versionedSymbol	test/test.d	52;"	v
w	test/test.d	15;"	v	struct:S
x	test/test.d	8;"	v
y	test/test.d	19;"	v	struct:S
z	test/test.d	14;"	v	struct:S
